sql: fix pagination in UPSERT

yuzefovich opened the pull request:
**sql: minor cleanup around upsert**

This commit removes a couple of duplicated "init" calls as well as some
unused parameters around upsert code.

Release note: None

**sql: fix pagination in UPSERT**

`optTableUpserter` was incorrectly overriding `curBatchSize` method by
returning `insertRows.Len`, but that container is not actually used
anywhere. As a result, `curBatchSize` was always considered 0, and we
didn't perform the pagination on the UPSERTs. The bug was introduced in
#33339 (in 19.1.0).

Fixes: #51391.

Release note (bug fix): Previously, CockroachDB could hit a "command is
too large" error when performing UPSERT operation with many values.
Internally, we attempt to perform such operation by splitting it into
"batches", but the batching mechanism was broken.

yuzefovich committed:
sql: fix pagination in UPSERT

`optTableUpserter` was incorrectly overriding `curBatchSize` method by
returning `insertRows.Len`, but that container is not actually used
anywhere. As a result, `curBatchSize` was always considered 0, and we
didn't perform the pagination on the UPSERTs. The bug was introduced in
33339 (in 19.1.0).

Release note (bug fix): Previously, CockroachDB could hit a "command is
too large" error when performing UPSERT operation with many values.
Internally, we attempt to perform such operation by splitting it into
"batches", but the batching mechanism was broken.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51608)
<!-- Reviewable:end -->

yuzefovich commented:
Does this deserve a unit test?

yuzefovich committed:
sql: minor cleanup around upsert

This commit removes a couple of duplicated "init" calls as well as some
unused parameters around upsert code.

Release note: None

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>  Wow, nice find!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51608#-:-MCiBrjw6zro8IimubLj:bp9ig1z)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten and @RaduBerinde)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
